src/main/java/sirius/db/jdbc/SmartQuery.java

```diff
@@ -806,16 +806,18 @@ private void appendToSELECT(Compiler c,
 
     private Compiler selectCount() {
         Compiler c = new Compiler(descriptor);
-        if (!fields.isEmpty()) {
-            c.getSELECTBuilder().append("SELECT COUNT(");
-            if (distinct) {
-                c.getSELECTBuilder().append("DISTINCT");
-            }
+        c.getSELECTBuilder().append("SELECT COUNT(");
+
+        if (!fields.isEmpty() && distinct) {
+            c.getSELECTBuilder().append("DISTINCT ");
+            appendFieldList(c, false);
+        } else if (fields.stream().count() == 1) {
             appendFieldList(c, false);
```

Comment:
`SELECT count(id)` should be equivalent to `SELECT count(*)`, right? Then I'd prefer to just use `count(*)` in all cases

Reply:
https://www.w3resource.com/sql/aggregate-functions/count-function.php nope, small difference

Reply:
Or throw, as the programmer probably expected another outcome when specifying a field? like forgot to add "distinct"

Reply:
`SELECT COUNT(id)` doesn't take `null` values into account while `SELECT COUNT(*)` considers all fields of the row in it's calculation. It's possible that a count on id returns a different (lower) number of rows than a count on *. I'd therefore recommend to distinguish between those two statements.